src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/generator/AbstractQueryGenerator.java

```diff
@@ -47,6 +47,17 @@ private String generateIsEqual(@NonNull Criteria criteria, @NonNull List<Pair<St
         return String.format("r.%s=@%s", subject, subject);
     }
 
+    private String generateGreaterThan(@NonNull Criteria criteria, @NonNull List<Pair<String, Object>> parameters) {
+        final String subject = this.getCriteriaSubject(criteria);
+
+        Assert.isTrue(criteria.getSubjectValues().size() == 1,
+                "GREATER_THAN should have only one subject value");
+
+        parameters.add(Pair.with(subject, criteria.getSubjectValues().get(0)));
```

Comment:
I am not very sure if it works well when date, as we will convert date type to getTime(), before store in database. like this `"date": 946656000000,`

Reply:
Now it can store in database like "date": 946656000000.